kafka/metrics.go

```diff
@@ -0,0 +1,105 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kafka
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const (
+	instrumentName = "github.com/elastic/apm-queue/kafka"
+
+	unitCount = "1"
+
+	messageProducedCounterKey = "message.produced"
```

Comment:
Should we prefix the metrics with the actual object that's producing them? For instance, this metric is generated by the producer, and `kafka.NewConsumer` won't affect these.

```suggestion
	producerMessagesProducedKey = "producer.messages.produced"
```

Reply:
Agreed

Reply:
Do we want this to be `producer` or `kafka`? Do you think it make sense to add a dimension to be able to filter for producer?